src/aws/sqs.ts

```diff
@@ -4,14 +4,40 @@ import * as AWS from "aws-sdk"
 
 export class SQS {
     readonly sqs: AWS.SQS
+    readonly sqsEndpoint: string
     readonly sqsUrl: string
+    readonly queueName: string
+
+    constructor(sqsUrl: string) {
+        const splitURL = sqsUrl.split("/");
+        const sqsEndpoint = splitURL[0] + "//" + splitURL[2];
+        const queueName = splitURL[splitURL.length - 1];
 
-    constructor(endpoint: string, sqsUrl:string) {
         this.sqs = new AWS.SQS({
             apiVersion: "2015-03-31",
-            endpoint,
-        })
-        this.sqsUrl = sqsUrl
+            endpoint: sqsEndpoint
+        });
+        this.sqsEndpoint = sqsEndpoint;
+        this.sqsUrl = sqsUrl;
+        this.queueName = queueName;
+    }
+
+    async createQueue(): Promise<any> {
+        try {
+            const params = {
+                QueueName: this.queueName,
+            };
+            await this.sqs.createQueue(params).promise();
+            console.log("DONE CREATING QUEUE!");
+        } catch(err) {
+            if (err.toString().indexOf("NetworkingError: connect ECONNREFUSED") >= 0) {
```

Comment:
If the service is remote, then the network error will not mean what it is describing. If using a local docker image, then the dependency should be handled at a service level.